common/beerocks/bcl/source/beerocks_os_utils.cpp

```diff
@@ -86,10 +87,15 @@ void os_utils::kill_pid(std::string path, std::string file_name)
 {
     int pid_out;
     if (is_pid_running(path, file_name, &pid_out)) {
-        std::cout << "kill_pid() kill pid=" << pid_out << std::endl;
+        LOG(DEBUG) << __FUNCTION__ << " SIGTERM pid=" << pid_out << std::endl;
         kill(pid_out, SIGTERM);
-        sleep(2); //TODO: find alternative solution
+        auto timeout = std::chrono::steady_clock::now() + std::chrono::seconds(15);
```

Comment:
Timeout value should be exported as an argument.
But overall, busy-wait is never a good idea.
Maybe sleep for short intervals inside the while loop?
(NOTE: `waitpid()` doesn't support timeouts [without ugly alarm/select workaround])